backend/src/models/giftTransfer.model.ts

```diff
@@ -13,7 +14,8 @@ export interface IGiftTransfer {
   giftCost?: number
 }
 
-type IGiftTransferDocument = IGiftTransfer & Document
+export type IGiftTransferDocument = IGiftTransfer & Document
+interface IGiftTransferModel<T extends Document> extends PaginateModel<T> {}
```

Comment:
I think this could be defined as `type`:
`type IGiftTransferModel<T extends Document> = PaginateModel<T>`